src/resources/api.ts

```diff
@@ -276,6 +362,7 @@ export class Api<SecurityDefs extends string> extends Base<ApiDetails> {
     SecurityDefinition
   >;
   private readonly security?: Record<SecurityDefs, string[]>;
+  private readonly cors?: boolean | CorsOptions;
```

Comment:
I wonder if its worth implementing the CorsApi as an extension of the main API. Then we might be able to implement some TS shenanigans so we can remove methods from the main API interface in the case that CORS is enabled e.g. `Omit<Api, "options">`.

Reply:
Interesting point.

I think we should now consider the following:
- do we even need options as a public method? As they are used for preflight requests anyway.
- if we do allow them, would the users handler override our cors one.

Reply:
going to do it within the membrane, resolving this.